src/task.js

```diff
@@ -70,51 +86,63 @@ class Succeed /*::<x,a>*/ extends Task /*::<x, a>*/ {
   value: a;
   */
   constructor(value/*:a*/) {
-    super(Succeed$prototype$fork)
+    super()
     this.value = value
   }
   fork(succeed/*:(a:a) => void*/, fail/*:(x:x) => void*/)/*:void*/ {
     succeed(this.value)
   }
 }
-const Succeed$prototype$fork = Succeed.prototype.fork
 
 class Fail /*::<x, a>*/ extends Task /*::<x, a>*/ {
   /*::
   error: x;
   */
   constructor(error/*:x*/) {
-    super(Fail$prototype$fork)
+    super()
     this.error = error
   }
   fork(succeed/*:(a:a) => void*/, fail/*:(x:x) => void*/)/*:void*/ {
     fail(this.error)
   }
 }
-const Fail$prototype$fork = Fail.prototype.fork
+
 
 class Sleep /*::<x, a:void>*/ extends Task /*::<x, void>*/ {
   /*::
   time: Time;
   */
   constructor(time/*:Time*/) {
-    super(Sleep$prototype$execute)
+    super()
     this.time = time
   }
-  execute(succeed/*:(a:a) => void*/, fail/*:(x:x) => void*/)/*:?Abort<void>*/ {
-    const id = setTimeout(succeed, this.time, void(0))
-    return () => clearTimeout(id)
+  fork(succeed/*:(a:a) => void*/, fail/*:(x:x) => void*/)/*:number*/ {
+    return setTimeout(succeed, this.time, void(0))
+  }
+  abort(id/*:number*/)/*:void*/ {
+    clearTimeout(id)
+  }
+}
+
+class RequestAnimationFrame /*::<x>*/ extends Task /*::<x, Time>*/ {
```

Comment:
I think I'd prefer for this to be called `RequestAnimationFrameTask` or something along those lines (I'd also be ok with `RAFTask`, I suppose). The name being identical to the `requestAnimationFrame` function (which there are two of now, too) is too confusing.